### ISSUE TITLE
SemanticsTester and its friends are not public

### ISSUE THREAD

**Issue by rzagelow** [flutter/flutter#45111]
## Use case
Unit testing semantic trees is difficult without using SemanticsTester, TestSemantics and hasSemantics() (all found in semantics_tester.dart). This is what flutter seems to do for testing semantics, but these classes are not public so our team is forced to copy this file into our own code base.

## Proposal

Please make semantics_tester.dart public so that we can use SemanticsTester, TestSemantics and hasSemantics(), etc for testing semantics. Thanks!

**Comment by abhaysood** [comment]
Any reason why this is not public? It's really useful.

**Comment by abhaysood** [comment]
Found the right way to do this. For anyone else looking for the same use [matcherSemantics](https://api.flutter.dev/flutter/flutter_test/matchesSemantics.html).

**Comment by iapicca** [comment]
> Found the right way to do this. For anyone else looking for the same use [matcherSemantics](https://api.flutter.dev/flutter/flutter_test/matchesSemantics.html).
> 
> I think this issue can be closed. @darrenaustin @iapicca 

Maybe we should also hear what the report has to say @rzagelow

**Comment by rzagelow** [comment]
This looks like it will work.. I'll report back in a couple days to confirm.. I do see what I think is a mistake in the documentation example where "matchesSemanticsNode" is referenced.. I think it should be "matchesSemantics".. here is the sample code I'm talking about:

```
final SemanticsHandle handle = tester.ensureSemantics();
expect(tester.getSemantics(find.text('hello')), matchesSemanticsNode(label: 'hello'));
handle.dispose();

```
 Sorry for the delayed response..

**Comment by rzagelow** [comment]
This worked perfect for me @abhaysood!.. thanks!.. we can safely close @iapicca!.. Thanks both of you!!

**Comment by darrenaustin** [comment]
Closing as `matchesSemanticsNode` works for this case.
